magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAcquisitionServiceImpl.java

```diff
@@ -0,0 +1,211 @@
+        if (repository.startsWith("github") || repository.startsWith("http") || repository.startsWith("git")) {
+          getGitRepository(repository);
+        } else {
+          copyLocalRepository(repository);
+        }
+      });
+  }
+
+  @Override
+  public List<PolicyContext> loadPolicies() {
+    if (policyConfig == null) {
+      LOGGER.error("Class have to be initialized first!");
+      return List.of();
+    }
+
+    var policyContexts = new ArrayList<PolicyContext>();
+
+    policyConfig.getRepositories()
+      .stream()
+      .map(repository -> repository.replace("~", System.getProperty("user.home")))
+      .forEach(repository -> {
+        String repositoryPath = getTargetProjectDirectoryPath(repository).toString();
+
+        HashMap<String, Rule> repositoryRulesMap = loadRulesFromRepository(repositoryPath);
+
+        var repositoryPolicies = loadPoliciesFromRepository(repositoryPath, repositoryRulesMap);
+        policyContexts.addAll(repositoryPolicies);
+      });
+
+    return policyContexts;
+  }
+
+  private HashMap<String, Rule> loadRulesFromRepository(String repositoryPath ) {
+    File rulesDirectory = new File(repositoryPath + "/rules");
+    var rules = new HashMap<String, Rule>();
+
+    if(rulesDirectory.exists()) {
+      for (File ruleFile : Objects.requireNonNull(rulesDirectory.listFiles())) {
+        try {
+          Rule yamlRule = YAML_MAPPER.readValue(ruleFile, Rule.class);
+          rules.put(yamlRule.getId(), yamlRule);
+          LOGGER.info("Successfully loaded rule {}", yamlRule.getId());
+        } catch (IOException yamlIOException) {
+          LOGGER.error(yamlIOException.getMessage());
+        }
+      }
+    } else {
+      LOGGER.error("Rules directory {} doesn't exists!", rulesDirectory);
+    }
+
+    return rules;
+  }
+
+  private ArrayList<PolicyContext> loadPoliciesFromRepository(String repositoryPath, HashMap<String, Rule> repositoryRulesMap) {
+    File policiesDirectory = new File(repositoryPath + "/policies");
+
+    var policiesContexts = new ArrayList<PolicyContext> ();
+
+    if(policiesDirectory.exists()) {
+      for (File policyRule : Objects.requireNonNull(policiesDirectory.listFiles())) {
+        var policy = loadPolicy(policyRule);
+
+        if(policy != null) {
+          loadPolicyRulesFromRulesMap(repositoryRulesMap, policy);
+          var policyMetadata = new PolicyMetadata(policyRule.getPath(), getRepoHashOrLocalRepositoryString(repositoryPath));
+          policiesContexts.add(new PolicyContext(policyMetadata, policy));
+        }
+      }
+    } else {
+      LOGGER.error("Policies directory {} doesn't exists!", policiesDirectory);
+    }
+
+    return policiesContexts;
+  }
+
+  private void loadPolicyRulesFromRulesMap(HashMap<String, Rule> repositoryRulesMap, Policy policy) {
+    policy.getRules().forEach(rule -> {
+      if (repositoryRulesMap.containsKey(rule.getId())) {
+        rule.set(repositoryRulesMap.get(rule.getId()));
```

Comment:
`Rule.set()` can be difficult to maintain in the future. I would prefer putting found rules to new list and then replacing entire `policy.rules` list with this new one. This way, if rule is not found, policy will still contain that rule with only ID set - not usable for further processing by analyzer